validation/src/main/java/io/micronaut/validation/validator/DefaultValidator.java

```diff
@@ -1052,6 +1052,7 @@ public void keyedValue(String nodeName, Object key, Object keyedValue) {
             }
         } finally {
             context.removeLast();
+            context.messageTemplate(null);
```

Comment:
Instead of setting to null shouldn't this reset it to what it was previously?

Reply:
Sure, We can do that. Thanks. I thought that since we are checking for the context's message template, if null found then we move on to annotation's default messages., I thought it would make sense to set the message template to null. Can you help me understand more on the scenarios, where resetting it to previous message wold help us?

Reply:
@agugan Truthfully I'm not sure, but setting it to what it was prior reduces the chances of breaking some existing use case

Reply:
@jameskleeh Sorry I took some timeoff for personal reasons. I have updated as per your suggestions. Please do let me know if any changes required.